train.py

```diff
@@ -135,7 +156,13 @@ def train(opt):
             last_step = 0
 
         try:
-            ret = model.load_state_dict(torch.load(weights_path), strict=False)
+            checkpoint = torch.load(weights_path)
+            if opt.apex and 'model' in checkpoint:
```

Comment:
If we are training with apex then we check whether the checkpoint includes `model`, `optimizer` and `amp` keys, which would mean it has already been trained with Apex. If it does not include these keys then we must load it as a normal checkpoint